File: 0169-majority-element/0169-majority-element.py
class Solution:
    
    def majorityElement(self, nums: List[int]) -> int:
        
        majority=nums[0]
        cnt=1
        
        for i in range(1,len(nums)):
            
            if(nums[i]!=majority):
                        if not cnt:
                                majority=nums[i]
                                cnt+=1
                        else:
                                cnt-=1
            
            
            else:
            
                    cnt+=1
                
        return majority

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
            int cnt=1;
            int temp = nums[0];
        
            for(int i=1;i<nums.size();i++){
                
                    if(temp!=nums[i]){
                        
                            cnt--;
                            if(cnt==0){
                                    
                                    temp=nums[i];
                                    cnt=1;
                            }
                        
                    }
                    else{
                        
                                cnt++;
                                
                        
                    }
                
                
            }
        
            return temp;
        
    }
};